src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3485,6 +3486,46 @@ internal void SymbolDeclaredEvent(Symbol symbol)
             EventQueue?.TryEnqueue(new SymbolDeclaredCompilationEvent(this, symbol.GetPublicSymbol()));
         }
 
+        internal override void SerializePdbEmbeddedCompilationOptions(BlobBuilder builder)
+        {
+            WriteValue("checked", Options.CheckOverflow.ToString());
+            WriteValue("nullable", Options.NullableContextOptions.ToString());
+            WriteValue("unsafe", Options.AllowUnsafe.ToString());
+            WriteValue("language-version", LanguageVersion.ToString());
+
+            var optimization = Options.OptimizationLevel == OptimizationLevel.Debug
+                ? (Options.DebugPlusMode ? "debug+" : "debug")
+                : "release";
+
+            WriteValue("optimization", optimization);
```

Comment:
Several of these strings are repeated between the C# compiler, VB compiler and the unit tests. Should we define them as constants even for our own usage?

Reply:
I think repeating them is fine, since technically that's what customers would have to do. 

Should we make them public const? That way our customers get the benefit that we do for not having to handwrite the strings in.

Reply:
> I think repeating them is fine, since technically that's what customers would have to do.

If I was a customer I would define a set of `const` so I didn't have to repeat the string everywhere. 

> Should we make them public const? That way our customers get the benefit that we do for not having to handwrite the strings in.

I would say "no" until we get more feedback that this is worth the cost of adding to the public API.

Reply:
K, adding to internal for now.